src/interceptors/fetch/index.ts

```diff
@@ -32,7 +33,21 @@ export class FetchInterceptor extends Interceptor<HttpRequestEventMap> {
 
     globalThis.fetch = async (input, init) => {
       const requestId = uuidv4()
-      const request = new Request(input, init)
+
+      /**
+       * @note Resolve potentially relative request URL
+       * against the present `location`. This is mainly
+       * for native `fetch` in JSDOM.
+       * @see https://github.com/mswjs/msw/issues/1625
+       */
+      const resolvedInput =
```

Comment:
My only concern is that this will force the relative request URL resolution even for environments where making a request to a relative URL is allowed (i.e. the browser).